DEFEDIT-1084 Fix issue with protobuf name mangling

ragnard-king opened the pull request:
The clojure.string/{upper-case,lower-case,capitalize} functions
provide no means to pass a Locale, and without doing so we will
experience the "Infamous Turkish Locale" problem.

This commit introduces utility functions that use the root locale
instead, and fixes usages in camel-snake-kebab and the protobuf name
mangling.

Fixes https://github.com/defold/editor2-issues/issues/1099

ragnard-king committed:
DEFEDIT-1084 Fix issue with protobuf name mangling

The clojure.string/{upper-case,lower-case,capitalize} functions
provide no means to pass a Locale, and without doing so we will
experience the "Infamous Turkish Locale" problem.

This commit introduces utility functions that use the root locale
instead, and fixes usages in camel-snake-kebab and the protobuf name
mangling.

matgis-king reviewed (changes requested):
Reviewed! Our protobuf name mangling handles some cases different than the reference implementation. We might want to address that.

ragnard-king committed:
DEFEDIT-1084 Review fixes

- more closely match protobuf camel-casing algorithm
- remove unused locale macros

ragnard-king committed:
DEFEDIT-1084 Review fix

- don't die on empty string
